### PR TITLE
#164376193 setup Testing environment and CI

### PR DESCRIPTION
**Pull request by malfahad**
#### What does this PR do?
- ensures that the testing environment with jest is setup and continuous integration with Travis is setup.

#### Description the task to be implemented
- setup test environment
- setup Travis
- setup hound
- add codeclimate and Travis badges for test coverage, maintainability and build status.

#### How can this be manually tested?
- Fetch this branch by running `git fetch origin ch-setup-tests-164376193`
- Checkout to this branch by running ` git checkout ch-setup-tests-164376193`
- Install dependencies by running `npm install`
- Start the local server by running `npm start`

[ #164376152](https://www.pivotaltracker.com/story/show/164376152)

### CODE REVIEW

File: .hound.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+eslint:
+  enabled: true
+  config_file: .eslintrc.json
+  fail_on_violations: true
+  ignore_file: .eslintignore
+scss:
+  enabled: false
```
Comment (kdaglas) [inline .hound.yml]: new line needed here

**Review by armstrongsouljah — APPROVED** [review]
LGTM!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    - node_modules
+
+before_script:
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
+script:
+  - yarn test:coverage
+after_script:
+  - ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
```
Comment (kdaglas) [inline .travis.yml]: and also here, always remember to add new lines at the end of your script

**Review by kdaglas — APPROVED** [review]
other than the new lines all looks good to me

**Pull request merged**